src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyStringComparison.cs

```diff
@@ -208,5 +210,35 @@ private static ParameterInfo GetParameterInfo(INamedTypeSymbol type, bool isArra
         {
             return ParameterInfo.GetParameterInfo(type, isArray, arrayRank, isParams);
         }
+
+        private static IEnumerable<IMethodSymbol> GetAccessibleMethodsWithSameNameAsTargetMethod(
+            IInvocationOperation invocationExpression,
+            IMethodSymbol targetMethod)
+        {
+            var invocationStart = invocationExpression.Syntax.GetLocation().SourceSpan.Start;
+
+            var methodsWithSameNameAsTargetMethod =
+                invocationExpression.SemanticModel?
+                    .LookupSymbols(
```

Comment:
I might just be misunderstanding how this method is used, but I noticed this wasn't returning protected methods--hence the extra call to `GetMembers` to get those. Is there a simpler way to handle this?

Reply:
I would suggest not using LookupSymbols here as it is generally an expensive method call.

Instead I would build on top of existing code and add a `Where` clause to filter out methods whose static-ness doesn't match OR [SemanticModel.IsAccessible](https://sourceroslyn.io/Microsoft.CodeAnalysis/R/bfd11df5ee9625f2.html) returns `false`.

Reply:
I've replaced the call to `LookupSymbols` with a `Where` clause that checks `IMethodSymbol.IsStatic` and `SemanticModel.IsAccessible`. 

How should the accessibility check be handled when the `SemanticModel` is null? Currently I have the filter set up to treat the method as if it's not accessible in this situation.